### PR TITLE
IDEX-3940:Add to index new files while renaming refactoring

### PR DESCRIPTION
**Pull request by vinokurig**
@skabashnyuk @vparfonov please review

### CODE REVIEW

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/event/CreateFileNodeEvent.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import com.google.gwt.event.shared.GwtEvent;
+
+import org.eclipse.che.ide.project.node.FileReferenceNode;
+
+/**
+ * Event that notifies of item created.
+ *
+ * @author Igor Vinokur
+ */
+public class CreateFileNodeEvent extends GwtEvent<CreateFileNodeEventHandler> {
```
Comment (azatsarynnyy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/event/CreateFileNodeEvent.java]: This event fired when node is already created so plz name it **FileNodeCreatedEvent** and **FileNodeCreatedHandler** respectively

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/event/DeleteNodesEvent.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import org.eclipse.che.ide.project.node.ResourceBasedNode;
+
+import java.util.List;
+
+/**
+ * Event that notifies of nodes deleted
+ *
+ * @author Igor Vinokur
+ */
+public class DeleteNodesEvent extends GwtEvent<DeleteNodesEventHandler> {
```
Comment (azatsarynnyy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/event/DeleteNodesEvent.java]: see https://github.com/codenvy/che-core/pull/685#discussion_r51446069
rename event and handler in the past tense

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/event/RenameNodeEvent.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import com.google.gwt.event.shared.GwtEvent;
+
+import org.eclipse.che.ide.project.node.ItemReferenceBasedNode;
+
+/**
+ * Event that notifies of renaming provided
+ *
+ * @author Igor Vinokur
+ */
+public class RenameNodeEvent extends GwtEvent<RenameNodeEventHandler> {
```
Comment (azatsarynnyy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/event/RenameNodeEvent.java]: see https://github.com/codenvy/che-core/pull/685#discussion_r51446069
rename event and handler in the past tense

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/event/CreateFileNodeEvent.java (hunk 1 above)
Comment (vinokurig) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/event/CreateFileNodeEvent.java]: done

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/event/DeleteNodesEvent.java (hunk 2 above)
Comment (vinokurig) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/event/DeleteNodesEvent.java]: done

Comment (vinokurig) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/event/DeleteNodesEvent.java]: done

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/event/RenameNodeEvent.java (hunk 3 above)
Comment (vinokurig) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/event/RenameNodeEvent.java]: done

**Comment by skabashnyuk** [comment]
ok. @vparfonov ?

**Comment by vparfonov** [comment]
ok

**Comment by vinokurig** [comment]
Need to rework according to new vfs events that will be implemented in IDEX-4055

**Pull request closed without merging**